cmake/thirdparty/SetupCUDA.cmake

```diff
@@ -121,6 +120,14 @@ else()
     set(CUDA_HOST_COMPILER ${CMAKE_C_COMPILER})
 endif()
 
+# Set PIE options to empty for PGI since it doesn't understand -fPIE This
+# option is set in the CUDA toolchain file so must be unset after
+# enable_language(CUDA)
```

Comment:
How would a user know to do this?  Can you instead remove the -fPIE from a flag from some variable instead?

Reply:
I couldn't find another way to do this. It's because the toolchain sets it that makes it hard to undo.

Reply:
We also call enable_language(CUDA) in this file... can this be handled without user intervention? I just have no idea how people are supposed to know to do this.

Reply:
I'm not sure I understand. This doesn't require user intervention - if they enable CUDA and are using PGI, this change ensures that option ends up empty.

Reply:
OH! i read that comment the wrong way.. i thought you were implying the user must unset this variable ....